client/src/app/core/repositories/agenda/agenda-item-repository.service.ts

```diff
@@ -194,4 +195,8 @@ export class AgendaItemRepositoryService extends BaseRepositoryWithActiveMeeting
         const payload: AgendaItemAction.UpdatePayload[] = items.map(item => ({ id: item.id, type: agendaType }));
         return this.sendBulkActionToBackend(AgendaItemAction.UPDATE, payload);
     }
+
+    protected pipeViewModels(viewModels: ViewAgendaItem[]): ViewAgendaItem[] {
+        return this.treeService.makeFlatTree(viewModels, 'weight', 'parent_id');
```

Comment:
I'm not sure why you need to wrap the models here. You can just set a new property `tree_weight` and `level` according to the sorting. This should be sufficient.

Reply:
Yes, but I can use the `treeService.makeFlatTree`-function like I did it here, if I can modify the original array...

Reply:
> if I can modify the original array

Reply:
Yes. I have wanted to use the existing functions.